target_finder/src/nodes/target_locator.cpp

```diff
@@ -43,92 +55,116 @@ using industrial_extrinsic_cal::ROSCameraObserver;
+  ROSListenerTransInterface *target_to_camera_TI_;
+  // listen:pullTransform(), read:loadPose(scene_, file) and write:saveCurrentPose(scene, file)
+  // scene is optional if file is not ""
+  // otherwise scene is used to create the name of the file
+  // full file path takes the form data_directory_/filexxx.yaml
 };
 
 TargetLocatorService::TargetLocatorService(ros::NodeHandle nh)
 {
   nh_ = nh;
   ros::NodeHandle pnh("~");
 
-  int rows, cols;
-  double diameter, spacing;
+  // In launch you may also set the dynamic reconfigure variables of:
+  // target_locator/target_rows
+  // target_locator/target_cols
+  // target_locator/target_circle_dia
+  // target_locator/target_spacing
+
   if (!pnh.getParam("image_topic", image_topic_))
   {
     ROS_ERROR("Must set param:  image_topic");
   }
 
-  if (!pnh.getParam("target_type", target_type_))
-  {
-    ROS_ERROR("Must set param: target_type");
-  }
-
-  if (!pnh.getParam("camera_name", camera_name_))
+  if (!pnh.getParam("camera_name", camera_name_)) // need this to get the intrinsics 
   {
     ROS_ERROR("Must set param: camera_name");
   }
 
   camera_observer_ = new ROSCameraObserver(image_topic_, camera_name_);
 
-  if (target_type_ == pattern_options::ModifiedCircleGrid)
+  camera_observer_->use_circle_detector_ = true;
+  pnh.getParam("use_circle_detector",  camera_observer_->use_circle_detector_);
+  
+  if (!pnh.getParam("target_frame", target_frame_)) // need this to get the intrinsics 
   {
-    if (!pnh.getParam("target_rows", target_rows_))
-    {
-      ROS_ERROR("Must set param:  target_rows");
-    }
-    if (!pnh.getParam("target_cols", target_cols_))
-    {
-      ROS_ERROR("Must set param:  target_cols");
-    }
-    if (!pnh.getParam("target_circle_dia", diameter))
-    {
-      ROS_ERROR("Must set param:  target_circle_dia");
-    }
-    if (!pnh.getParam("target_spacing", spacing))
-    {
-      ROS_ERROR("Must set param:  target_spacing");
-    }
-    initMCircleTarget(target_rows_, target_cols_, diameter, spacing);
-    pnh.getParam("publish_target_marker", target_->pub_rviz_vis_);
+    ROS_ERROR("Must set param: target_frame");
   }
-  else
+
+  if (!pnh.getParam("camera_frame", camera_frame_)) // need this to get the intrinsics 
   {
-    ROS_ERROR("Only target_type = 2 ModifiedCircleGrid targets supported");
+    ROS_ERROR("Must set param: camera_frame");
   }
 
-  std::string service_name;
-  if (!pnh.getParam("service_name", service_name))
+  if (!pnh.getParam("data_directory", data_directory_)) // need this to get the intrinsics 
   {
-    service_name = "TargetLocateService";
+    ROS_ERROR("Must set param: data_directory");
   }
-  target_locate_server_ = nh_.advertiseService(service_name.c_str(), &TargetLocatorService::executeCallBack, this);
+
+  // initialize the transform interface it listens from the frame in the constructor to the reference frame
+  target_to_camera_TI_ = new industrial_extrinsic_cal::ROSListenerTransInterface(target_frame_);
```

Comment:
Same thing here, this raw pointer should probably be `delete`d in the destructor just to be careful.